main.go

```diff
@@ -17,34 +18,58 @@ import (
 	"os/exec"
 	"regexp"
 	"strings"
+	"sync"
+	"time"
 )
 
+type output struct {
+	content string
+	mu      *sync.Mutex
+}
+
 // Cmd is typically constructed through the Command() call and provides state
 // to the execution engine.
 type Cmd struct {
 	cmd       *exec.Cmd
 	env       []string
 	exitError error
-	executed  bool
-	stdout    string
-	stderr    string
+	status    string
```

Comment:
The `executed` flag was simple because all we cared about was to know if a command  had been executed, or not. With long-running programs, we have a few more states. Off the top of my head:
- initialized
- started
- running
- finished

I would think about all the possible states and how they can be applied to both long-running and short-lived commands.

Reply:
I think started can be derived from the state being either running or finished, which would leave us with three types of state that can be recorded on `Command`:
 * initialized
 * running - only relevant for long running commands. 
 * finished